enterprise/server/auth/auth.go

```diff
@@ -243,25 +245,40 @@ func extractToken(issuer, slug string, idToken *oidc.IDToken) (*userToken, error
 }
 
 func (a *oidcAuthenticator) getIssuer() string {
+	if err := a.initialize(); err != nil {
+		return ""
+	}
 	return a.issuer
 }
 
 func (a *oidcAuthenticator) getSlug() string {
+	if err := a.initialize(); err != nil {
+		return ""
+	}
 	return a.slug
 }
 
 func (a *oidcAuthenticator) authCodeURL(state string, opts ...oauth2.AuthCodeOption) string {
+	if err := a.initialize(); err != nil {
+		return ""
+	}
 	return a.oauth2Config.AuthCodeURL(state, opts...)
 }
 
 func (a *oidcAuthenticator) exchange(ctx context.Context, code string, opts ...oauth2.AuthCodeOption) (*oauth2.Token, error) {
+	if err := a.initialize(); err != nil {
+		return nil, err
+	}
 	return a.oauth2Config.Exchange(ctx, code, opts...)
 }
 
 func (a *oidcAuthenticator) verifyTokenAndExtractUser(ctx context.Context, jwt string, checkExpiry bool) (*userToken, error) {
+	if err := a.initialize(); err != nil {
+		return nil, err
+	}
 	conf := a.oidcConfig
 	conf.SkipExpiryCheck = !checkExpiry
-	validToken, err := a.provider.Verifier(conf).Verify(ctx, jwt)
+	validToken, err := a.Provider.Verifier(conf).Verify(ctx, jwt)
```

Comment:
I'm maybe missing other occurrences -- but if this is the only place a method is called on provider -- maybe we could do the providerOnce.Do(init()) here and then we would not need initialize() elsewhere?